File: Src/Icm.Core/EventArgs/ErrorEventArgs.vb
Namespace Icm

    Public Class ErrorEventArgs(Of T)
        Inherits EventArgs(Of T)

        Private exception_ As Exception
        Public Property Exception() As Exception
            Get
                Return exception_
            End Get
            Set(ByVal value As Exception)
                exception_ = value
            End Set
        End Property

        Public Sub New(ByVal _data As T, ByVal _exception As Exception)
            MyBase.New(_data)
            exception_ = _exception
        End Sub

    End Class

End Namespace

File: Src/Icm.Core/EventArgs/ErrorEventArgs.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace Icm
{

	public class ErrorEventArgs<T> : EventArgs<T>
	{

		private Exception exception_;
		public Exception Exception {
			get { return exception_; }
			set { exception_ = value; }
		}

		public ErrorEventArgs(T _data, Exception _exception) : base(_data)
		{
			exception_ = _exception;
		}

	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
